app/validation/validator.py

```diff
@@ -122,80 +122,79 @@ def validate_answer(self, block, answer):
-                    used_answer_decimals = int(answer.get('decimal_places', 0))
-
-        if not used_answer_exists:
-            error_message = '{} used for {} is not an answer id in schemas'.format(used_answer_id, answer_id)
-            range_errors.append(self._error_message(error_message))
-        elif used_answer_type not in ['Number', 'Currency', 'Percentage']:
-            error_message = '{} is of type {} and therefore can not be passed to max/min values for {}'\
-                .format(used_answer_id, used_answer_type, answer_id)
-            range_errors.append(self._error_message(error_message))
-        elif used_answer_decimals > answer_decimals:
-            if answer_decimals == 0:
-                error_message = '{} of type decimal is being passed to ' \
-                                'max/min value for {} of type integer'.format(used_answer_id, answer_id)
-            else:
-                error_message = '{} is of type decimal with {} places is being passed to' \
-                                ' max/min value for {} of {} decimal_places'\
-                    .format(used_answer_id, used_answer_decimals, answer_id, answer_decimals)
-
-            range_errors.append(self._error_message(error_message))
-
-        return range_errors
-
-    def validate_range_value(self, value, answer_id, answer_decimals):
-        error_message = 'Decimal Places used in {} should be less than or equal to {}, currently {}' \
-            .format(answer_id, MAX_DECIMAL_PLACES, answer_decimals)
-
-        if answer_decimals > MAX_DECIMAL_PLACES:
-            return [self._error_message(error_message)]
-
-        error_message = 'Value {} used in {} should be between system limits {} to {}'\
-            .format(value, answer_id, MIN_NUMBER, MAX_NUMBER)
-
-        if MIN_NUMBER <= value <= MAX_NUMBER:
-            return []
-
-        return [self._error_message(error_message)]
+    def _get_answer_range_values(self, answer, answer_ranges):
+
+        return {
+            'min': self._get_answer_minimum_or_maximum(answer['id'],
+                                                       answer_ranges,
+                                                       answer.get('min_value', {'value': 0}),
+                                                       answer.get('decimal_places', 0),
+                                                       'minimum'),
+            'max': self._get_answer_minimum_or_maximum(answer['id'],
+                                                       answer_ranges,
+                                                       answer.get('max_value', {'value': MAX_NUMBER}),
+                                                       answer.get('decimal_places', 0),
+                                                       'maximum'),
+            'decimal_places': answer.get('decimal_places', 0),
+        }
+
+    def _get_answer_minimum_or_maximum(self, answer_id, answer_ranges, attributes, decimal_places, min_or_max):
```

Comment:
`attributes` here is not very clear as to what it actually is

Reply:
It is basically the attribute of whether the answer has a min_value or a max_value

Reply:
but the name `attributes` is too vague. i had to look at the calling method to have any idea what it was.

Reply:
how about `min_max_attribute`?

Reply:
Im not really a fan of the whole min/max in the names of thing. you are naming the function based on the values you are passing in and not on what it actually does

Reply:
So the attributes being passed in can vary from values, to a dict, so maybe something like `range_attributes`

Reply:
if what gets passed in car vary in type then that is also bad